ledger-service/http-json/src/main/scala/com/digitalasset/http/util/Logging.scala

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http.util
+
+import com.daml.logging.LoggingContextOf
+import com.daml.logging.LoggingContextOf.{label, newLoggingContext, withEnrichedLoggingContext}
+
+import java.util.UUID
+
+object Logging {
+
+  abstract case class CorrelationID()
+  abstract case class RequestID()
+
+  def correlationIdLogCtx[Z](fn: LoggingContextOf[CorrelationID] => Z): Z =
+    newLoggingContext(label[CorrelationID], Map("CorrelationID" -> UUID.randomUUID().toString))(fn)
+
+  def extendWithRequestIdLogCtx[Z](
+      fn: LoggingContextOf[CorrelationID with RequestID] => Z
+  )(implicit lc: LoggingContextOf[CorrelationID]): Z =
+    withEnrichedLoggingContext(label[RequestID], Map("RequestID" -> UUID.randomUUID().toString))
+      .run(fn)
```

Comment:
Other components in the stack use `snake_case` in the logging context, we might want to stick to that.
```suggestion
  def correlationIdLogCtx[Z](fn: LoggingContextOf[CorrelationID] => Z): Z =
    newLoggingContext(label[CorrelationID], Map("correlation_id" -> UUID.randomUUID().toString))(fn)

  def extendWithRequestIdLogCtx[Z](
      fn: LoggingContextOf[CorrelationID with RequestID] => Z
  )(implicit lc: LoggingContextOf[CorrelationID]): Z =
    withEnrichedLoggingContext(label[RequestID], Map("request_id" -> UUID.randomUUID().toString))
      .run(fn)
```